components/writer-reload/element.js

```diff
@@ -0,0 +1,48 @@
+import { LitElement, nothing } from "lit";
+
+export default class MDNWriterReload extends LitElement {
+  connectedCallback() {
+    super.connectedCallback();
+    this._pollForChanges();
+  }
+
+  render() {
+    return nothing;
+  }
+
+  async _reloadIfChanged() {
+    const url = new URL(
+      "index.json",
+      new URL(location.pathname + "/", location.origin),
+    );
+    const res = await fetch(url);
+    if (res.ok) {
+      const state = await res.text();
+      if (!this._state) {
+        this._state = state;
+      } else if (this._state !== state) {
+        location.reload();
+      }
+    } else {
+      console.error("Failed to fetch document", res.status, res.statusText);
+    }
+  }
+
+  async _pollForChanges() {
+    while (true) {
+      if (document.visibilityState === "visible") {
+        await this._reloadIfChanged();
+      }
```

Comment:
Nit: Shouldn't we only poll for changes while the page is visible, i.e. use the [visibilitychange](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event) to toggle polling on and off? Admittedly the performance impact is probably negligible.

Reply:
Yeah, I can't imagine it's a performance problem, and the logic becomes significantly more complex: we have to add state to check if we're currently polling, good chance for race conditions, etc.